src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -30,7 +28,10 @@ public static PineconeConnection createIndexIfNotExistsDataPlane(int dimension,
 
         // Do not proceed until the newly created index is ready
         isIndexReady(indexName, controlPlaneClient);
-        return new PineconeConnection(apiKey, indexName);
+
+        // Adding to test PineconeConnection(pineconeConfig, indexName) constructor
+        PineconeConfig config = new PineconeConfig(apiKey);
+        return new PineconeConnection(config, indexName);
```

Comment:
@jhamon this is where we are using the new constructor for all of the data plane tests. If the indexes don't exist, we create it using the new PineconeConfig by passing config object and indexName.